twisted/conch/ssh/keys.py

```diff
@@ -782,18 +783,23 @@ def public(self):
         return Key(self._keyObject.public_key())
 
 
-    def fingerprint(self):
+    def fingerprint(self, hash_function=sha256):
```

Comment:
In OpenSSH the known_hosts model uses the complete public key and not the fingerprint. As you surmised, in general the fingerprint would be used as a shorthand to allow users to verify it manually (although it's possible somebody, somewhere is inappropriately using a fingerprint for programmatic comparison). I agree that the standard deprecation process should be fine.